lib/edools/lesson_progress.rb

```diff
@@ -0,0 +1,148 @@
+      raise ArgumentError, 'missing enrollment_id' if enrollment_id.nil?
+
+      url = "#{base_url_enrollments}/#{enrollment_id}/lessons_progresses"
+      data = { enrollment_id: enrollment_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
+        lessons_progresses: objectify_lessons_progresses(response[:lessons_progresses]),
+        paginate: objectify_paginate(response)
+      )
+    end
+
+    def self.find_paginated_from_students(student_id: , page: 1, per_page: 50)
+      raise ArgumentError, 'missing student_id' if student_id.nil?
+
+      url = "#{base_url_students}/#{student_id}/lessons_progresses"
+      data = { student_id: student_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
+        lessons_progresses: objectify_lessons_progresses(response[:lessons_progresses]),
+        paginate: objectify_paginate(response)
+      )
+    end
+
+    def self.find_paginated_from_school_products(school_product_id: , page: 1, per_page: 50)
+      raise ArgumentError, 'missing school_product_id' if school_product_id.nil?
+
+      url = "#{base_url_school_products}/#{school_product_id}/lessons_progresses"
+      data = { school_product_id: school_product_id, page: page, per_page: per_page }
+
+      response = Edools::ApiRequest.request(:get, url, data)
+
+      OpenStruct.new(
+        lessons_progresses: objectify_lessons_progresses(response[:lessons_progresses]),
+        paginate: objectify_paginate(response)
+      )
+    end
+
+    def self.show(data = {})
+      raise ArgumentError, 'missing id' unless data.key? :id
+
+      response = Edools::ApiRequest.request(:get, base_url, data)
+
+      response[:lessons_progresses].map do |lesson_progress|
+        new lesson_progress
+      end
+    end
+
+    def self.base_url_enrollments
+      "#{Edools.base_url}/enrollments"
+    end
+
+    def self.base_url_students
+      "#{Edools.base_url}/students"
+    end
+
+    def self.base_url_school_products
+      "#{Edools.base_url}/school_products"
+    end
+
+    def self.base_url
+      "#{Edools.base_url}/lessons_progresses"
+    end
+
+    private
+
+    def self.objectify_paginate(paginate)
+      return paginate if paginate.nil?
+      OpenStruct.new(
+        current_page: paginate.dig(:current_page),
+        per_page: paginate.dig(:per_page),
+        total_pages: paginate.dig(:total_pages),
+        total_count: paginate.dig(:total_count)
+      )
+    rescue NoMethodError => error
+      puts Edools.logger.error(error)
+    end
+
+    def self.objectify_lessons_progresses(lessons_progresses)
+      lessons_progresses.map do |lesson_progress|
+        objectify_lesson_progress(lesson_progress)
+      end
+    end
+
+    def self.objectify_lesson_progress(lesson_progress)
+      return lesson_progress if lesson_progress.nil?
```

Comment:
If we can have `nil` inside the array maybe we should compact the array created inside `self.objectify_lessons_progresses`?